impc_etl/jobs/parse/gene_production_status_parser.py

```diff
@@ -0,0 +1,196 @@
+        if col_name in rename_map:
+            gentar_df = gentar_df.withColumnRenamed(col_name, rename_map[col_name])
+
+    df = gentar_df
+
+    def _create_status_map(d: Dict):
+        return spark.createDataFrame([
+            {"src_production_status": k, "dst_production_status": v} for k, v in d.items()
+        ])
+
+    def map_status(df, d: Dict, col_name: str):
+        map_df = _create_status_map(d)
+        df = df.join(map_df, lower(col(col_name)) == lower(col("src_production_status")), "left_outer")
+        df = df.withColumn(col_name,
+            when(col("dst_production_status").isNotNull() & (col("dst_production_status") != "NULL"), col("dst_production_status"))
+            .when(col("dst_production_status") == "NULL", lit(None))
+            .otherwise(col(col_name))
+        ).drop("src_production_status", "dst_production_status")
+        return df
+
+    def collapse_status(df, d: Dict, cols: List[str], target: str):
+        df = df.withColumn(target, lit(None).cast(StringType()))
+        rank_udf = udf(lambda x: list(d.values()).index(x) if x else 0, IntegerType())
+        for src in cols:
+            map_df = _create_status_map(d)
+            df = df.join(map_df, lower(col(src)) == lower(col("src_production_status")), "left_outer")
+            df = df.withColumn(target,
+                when(col(target).isNull() & col("dst_production_status").isNotNull(), col("dst_production_status"))
+                .when(rank_udf("dst_production_status") > rank_udf(target), col("dst_production_status"))
+                .otherwise(col(target))
+            ).drop("src_production_status", "dst_production_status")
+        return df
+
+    mouse_status_map = {
+        "Chimeras obtained": "Assigned for Mouse Production and Phenotyping",
+        "Micro-injection in progress": "Assigned for Mouse Production and Phenotyping",
+        "Mouse Allele Modification Genotype Confirmed": "Mice Produced",
+        "Rederivation Complete": "Mice Produced",
+        "Rederivation Started": "Mice Produced",
+        "Cre Excision Started": "Started",
+        "Cre Excision Complete": "Started",
+        "Genotype confirmed": "Mice Produced",
+        "Phenotype Attempt Registered": "Mice Produced",
+    }
+
+    es_status_map_1 = {
+        "Aborted - ES Cell QC Failed": "Not Assigned for ES Cell Production",
+        "Assigned - ES Cell QC In Progress": "Assigned for ES Cell Production",
+        "Assigned - ES Cell QC Complete": "ES Cells Produced",
+    }
+
+    df = collapse_status(df, mouse_status_map, [
+        "null_allele_production_status", "conditional_allele_production_status",
+        "crispr_allele_production_status", "crispr_conditional_allele_production_centre"
+    ], "mouse_production_status")
+
+    df = collapse_status(df, es_status_map_1, [
+        "null_allele_production_status", "conditional_allele_production_status"
+    ], "es_cell_production_status")
+
+    es_status_map_2 = {
```

Comment:
The original version had a more meaningful variable name: `allele_es_cells_prod_status_map`